sdk/communication/Azure.Communication.Administration/src/PhoneNumberAdministrationClient.cs

```diff
@@ -22,31 +22,47 @@ public class PhoneNumberAdministrationClient
         internal PhoneNumberAdministrationRestClient RestClient { get; }
 
         /// <summary>
-        /// Initializes a phone number administration client with an Azure resource connection string.
+        /// Initializes a phone number administration client with an Azure resource connection string and optional client options.
         /// </summary>
-        public PhoneNumberAdministrationClient(string connectionString)
-            : this(new PhoneNumberAdministrationClientOptions(), ConnectionString.Parse(connectionString))
+        public PhoneNumberAdministrationClient(string connectionString, PhoneNumberAdministrationClientOptions? options = default)
+            : this(
+                  options ?? new PhoneNumberAdministrationClientOptions(),
+                  ConnectionString.Parse(connectionString))
         { }
 
         /// <summary>
-        /// Initializes a phone number administration client with an Azure resource connection string and client options.
+        /// Initializes a phone number administration client with a token credential.
         /// </summary>
-        public PhoneNumberAdministrationClient(string connectionString, PhoneNumberAdministrationClientOptions? options = default)
+        public PhoneNumberAdministrationClient(Uri endpoint, TokenCredential tokenCredential, PhoneNumberAdministrationClientOptions? options = default)
             : this(
+                  endpoint,
                   options ?? new PhoneNumberAdministrationClientOptions(),
-                  ConnectionString.Parse(connectionString))
+                  tokenCredential)
         { }
 
-        internal PhoneNumberAdministrationClient(PhoneNumberAdministrationClientOptions options, ConnectionString connectionString)
+        private PhoneNumberAdministrationClient(PhoneNumberAdministrationClientOptions options, ConnectionString connectionString)
             : this(new ClientDiagnostics(options), options.BuildHttpPipeline(connectionString), connectionString.GetRequired("endpoint"))
```

Comment:
It feels more natural to put important arg first, connectionString.

Reply:
This is done to avoid naming conflicts.

Reply:
this is a bit tricky, but I had to do it this way to avoid conflict with the public constructor with the same argument that cannot control nullability.